File: nodejs/test/topjslibs/GoogleSearchResultTest.ts
import * as assert from "assert";
import {describe} from "mocha";
import {GoogleSearchResult} from "../../src/topjslibs/GoogleSearchResult";
import * as util from "./util";

describe("GoogleSearchResultTest", () => {
  describe("getLinks()", () => {
    it("should get the first 5 links from the Google search result", () => {
      // Arrange
      const expected = [
        "https://www.test.de/",
        "https://www.test.de/shop/test-hefte/",
        "https://www.test.de/thema/",
        "https://www.oekotest.de/",
        "https://de.wikipedia.org/wiki/Test_(Zeitschrift)",
      ];
      const fetchHtmlLambda = () => new Promise<string>((resolve) => resolve(util.getHtmlFrom("google.com.html")));

      return new GoogleSearchResult(5, fetchHtmlLambda)
      // Act
      .getLinks()
      // Assert
      .then((actual) => assert.deepStrictEqual(actual, expected));
    });
  });
});

File: rust/src/top_js_libs/google_search_result_test.rs
use crate::top_js_libs::google_search_result::GoogleSearchResult;
use crate::top_js_libs::test_util::get_html_from;

#[test]
fn get_links() {
    // Arrange
    let google_search_result = GoogleSearchResult::new(5, Box::from(|| get_html_from("google.com.html")));
    let mut expected = Vec::new();
    expected.push("https://www.test.de/".to_string());
    expected.push("https://www.test.de/shop/test-hefte/".to_string());
    expected.push("https://www.test.de/thema/".to_string());
    expected.push("https://www.oekotest.de/".to_string());
    expected.push("https://de.wikipedia.org/wiki/Test_(Zeitschrift)".to_string());

    // Act
    let actual = google_search_result.get_links();

    // Assert
    assert_eq!(expected, actual);
}